pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -244,6 +242,41 @@ func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDe
 		if err != nil {
 			return "", false, err
 		}
+		if len(availLeaseChars) == 0 {
+			logger.Warn("no GCP MachinePoolNameLease characters available, setting condition")
+			conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+				pool.Status.Conditions,
+				hivev1.NoMachinePoolNameLeasesAvailable,
+				corev1.ConditionTrue,
+				"AllCharactersInUse",
+				"All characters already in use for GCP MachinePoolNameLeases",
+				controllerutils.UpdateConditionIfReasonOrMessageChange,
+			)
+			if changed {
+				pool.Status.Conditions = conds
+				if err := a.client.Status().Update(context.Background(), pool); err != nil {
+					logger.WithError(err).Error("failed to update MachinePool conditions")
+				}
+			}
+			return "", false, fmt.Errorf("no lease chars available, unable to assign MachinePoolNameLease to pool %s", pool.Name)
```

Comment:
Should this return an error? I think it would be better to watch machinepoolnameleases in the controller and requeue the machinepools on changes to leases.